server/src/main/kotlin/suwayomi/tachidesk/manga/impl/util/GetComicInfo.kt

```diff
@@ -15,46 +15,56 @@ import suwayomi.tachidesk.manga.model.table.MangaTable
 import uy.kohesive.injekt.Injekt
 import uy.kohesive.injekt.api.get
 import java.nio.file.Path
+import java.time.Instant
+import java.time.ZoneId
 import kotlin.io.path.deleteIfExists
 import kotlin.io.path.div
 import kotlin.io.path.outputStream
 
-/**
- * Creates a ComicInfo instance based on the manga and chapter metadata.
- */
 fun getComicInfo(
     manga: ResultRow,
     chapter: ResultRow,
     chapterUrl: String,
     categories: List<String>?,
-) = ComicInfo(
-    title = ComicInfo.Title(chapter[ChapterTable.name]),
-    series = ComicInfo.Series(manga[MangaTable.title]),
-    number =
-        chapter[ChapterTable.chapter_number].takeIf { it >= 0 }?.let {
-            if ((it.rem(1) == 0.0f)) {
-                ComicInfo.Number(it.toInt().toString())
-            } else {
-                ComicInfo.Number(it.toString())
-            }
-        },
-    web = ComicInfo.Web(chapterUrl),
-    summary = manga[MangaTable.description]?.let { ComicInfo.Summary(it) },
-    writer = manga[MangaTable.author]?.let { ComicInfo.Writer(it) },
-    penciller = manga[MangaTable.artist]?.let { ComicInfo.Penciller(it) },
-    translator = chapter[ChapterTable.scanlator]?.let { ComicInfo.Translator(it) },
-    genre = manga[MangaTable.genre]?.let { ComicInfo.Genre(it) },
-    publishingStatus =
-        ComicInfo.PublishingStatusTachiyomi(
-            ComicInfoPublishingStatus.toComicInfoValue(manga[MangaTable.status].toLong()),
-        ),
-    categories = categories?.let { ComicInfo.CategoriesTachiyomi(it.joinToString()) },
-    inker = null,
-    colorist = null,
-    letterer = null,
-    coverArtist = null,
-    tags = null,
-)
+): ComicInfo {
+    val dateUpload = chapter[ChapterTable.date_upload]
+    val localDate =
+        dateUpload.let {
+            Instant.ofEpochMilli(it).atZone(ZoneId.systemDefault()).toLocalDate()
+        }
```

Comment:
```suggestion
        Instant.ofEpochMilli(dateUpload).atZone(ZoneId.systemDefault()).toLocalDate()
```
Please validate that we store them as Milliseconds as well, we store a lot of things as Seconds instead so this might need to be changed

Reply:
I tested the code and I get the right date when using milisec.